### PR TITLE
Add Gemm sizes check in the backward data tuning parameters pass

### PR DESCRIPTION
**Pull request by yiqian1**

### CODE REVIEW

File: mlir/lib/Dialect/MIOpen/Tuning/GridwiseGemmParams.cpp [hunk 1]
```diff
@@ -20,6 +20,13 @@ LogicalResult PopulateParams::populateDerived(
     return failure();
   }
```
Comment (jerryyin) [inline mlir/lib/Dialect/MIOpen/Tuning/GridwiseGemmParams.cpp:22]: You should make the parallel change in `PopulateParamsXDL::populateDerived`

**Comment by yiqian1** [comment]
@ltqin @asleepzzz Please review this PR. These changes address [issue#193](https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/193), moving gemmSize checks from MIOpen to MLIR. MIOpen changes will be in another PR.

**Comment by whchung** [comment]
@asleepzzz / @ltqin need your review on this PR.

File: mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h [hunk 2]
```diff
@@ -846,6 +846,24 @@ class PopulateParamsXDL : public PopulateParamsBase {
...
+    auto gemmN = gemmSize.gemmN;
+    auto gemmK = gemmSize.gemmK;
+
+    // unsupported xdlops-gemm
+    if (gemmM % 16 != 0 && gemmN % 64 != 0)
+      return failure();
+
+    const auto waveSize = 64;
+
+    if ((gemmM * gemmN) % 256 == 0 && (gemmK * gemmM) % waveSize == 0 &&
```
Comment (asleepzzz) [inline mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h:860]: there are a upper bound : gemmM * gemmN /waveSize

Comment (asleepzzz) [inline mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h:864]: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/b0956812043e5d8ff2279eb23eadd1419e4c07a7/src/solver/conv_hip_implicit_gemm_xdlops_common.cpp#L131

    const auto BlockSize = GemmNPerBlock * GemmMPerBlock / (GemmMPerWave * GemmNPerWave) * WaveSize;

    // fail with blockSize >= 512

Comment (yiqian1) [reply mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h:864]: This function is a copy of MIOpen code:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/include/miopen/solver/implicitgemm_util.hpp#L622

**Review by asleepzzz — APPROVED** [review]

**Pull request merged**